src/dotnet/ShellShimMaker/WindowsEnvironmentPath.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ShellShimMaker
+{
+    internal class WindowsEnvironmentPath : IEnvironmentPath
+    {
+        private readonly IReporter _reporter;
+        private readonly IEnvironmentProvider _environmentProvider;
+        private const string PathName = "PATH";
+        private readonly string _packageExecutablePath;
+
+        public WindowsEnvironmentPath(
+            string packageExecutablePath, IReporter reporter,
+            IEnvironmentProvider environmentProvider)
+        {
+            _packageExecutablePath 
+                = packageExecutablePath ?? throw new ArgumentNullException(nameof(packageExecutablePath));
+            _environmentProvider
+                = environmentProvider ?? throw new ArgumentNullException(nameof(environmentProvider));
+            _reporter
+                = reporter ?? throw new ArgumentNullException(nameof(reporter));
+        }
+
+        public void AddPackageExecutablePathToUserPath()
+        {
+            if (PackageExecutablePathExists()) return;
+
+            var existingUserEnvPath = Environment.GetEnvironmentVariable(PathName, EnvironmentVariableTarget.User);
```

Comment:
Why not use `_environmentProvider`?

Reply:
_environmentProvider does not have _GetEnvironmentVariable(PathName, EnvironmentVariableTarget.User)_ which is a netcore2.0 new added API to change registry.

Project where _environmentProvider from are netstandard

Reply:
Got it.